### PR TITLE
Add Auth emulator tests 🔐

### PR DESCRIPTION
**Pull request by jhuleatt**
### Description

Now that the Auth emulator has been released, remove the mocked out auth implementation we had been using in tests, and use the Auth emulator instead!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -69,11 +70,11 @@
     "globalthis": "^1.0.1",
     "husky": "^4.3.0",
     "node-fetch": "^2.6.1",
-    "react": "^0.0.0-experimental-4ead6b530",
-    "react-dom": "^0.0.0-experimental-4ead6b530",
+    "react": "experimental",
```
Comment (jhuleatt) [inline package.json:73]: need to add firebase-tools to `devDependencies`

File: test/auth.test.tsx [hunk 2]
```diff
@@ -1,146 +1,155 @@
...
+import { baseConfig } from './appConfig';
+
+/**
+ * These tests have to be skipped in CI because of an issue with jsdom.
+ *
+ * The Auth emulator server sets `access-control-allow-headers: *`
+ * jsdom doesn't currently work with that setting: https://github.com/jsdom/jsdom/issues/2408#issuecomment-480969435
+ *
+ * To fix this when running tests manually, modify `node_modules/jsdom/lib/jsdom/living/xhr-utils.js`
+ * with the changes in https://github.com/jsdom/jsdom/pull/3073
```
Comment (jhuleatt) [inline test/auth.test.tsx:16]: FYI

Comment (jhuleatt) [reply test/auth.test.tsx:16]: This `skip` can be removed once this PR is merged: https://github.com/jsdom/jsdom/pull/3073

**Review by jamesdaniels — APPROVED** [review]

**Comment by jamesdaniels** [comment]
LGTM, just update w/master

**Pull request merged**